File: C++/Does array represent Heap.cpp
class Solution{
    public:
    bool isMaxHeap(int arr[], int n)
    {   
        for(int i=0;i<n;i++)
        {
            int l=2*i + 1;
            int r=2*i + 2;
            if(l<n)
            {
                if(arr[l]>arr[i])
                    return 0;
            }
            if(r<n)
            {
                if(arr[r]>arr[i])
                    return 0;
            }
        }
        return 1;
    }
};

File: JAVA/Does array represent Heap.java
class Solution {
    
    public boolean countSub(long arr[], long n)
    {
        // Your code goes here
        int j = 0;
        for(int i = 0;i <n  ;i++)
        {
             if(j +i + 2 < n)
              {
         if( arr[i] <arr[i + j +2])
                return false;
              }
              if(j +i + 1 < n)
              {
         if( arr[i] < arr[i + j + 1] )
                
               return false;
              } 
                
                j++;
        }
        
        return true;
    }
}
